web/packages/api/src/toEthereum_v2.ts

```diff
@@ -1198,3 +1341,99 @@ async function buildMessageId(
     ])
     return blake2AsHex(entropy)
 }
+
+function createPNASourceParachainTx(
+    parachain: ApiPromise,
+    ethChainId: number,
+    assetHubParaId: number,
+    sourceAccount: string,
+    asset: Asset,
+    beneficiaryAccount: string,
+    amount: bigint,
+    totalFeeInDot: bigint,
+    messageId: string
+): SubmittableExtrinsic<"promise", ISubmittableResult> {
+    const assets = {
+        v4: [
+            {
+                id: asset.location,
+                fun: { Fungible: amount },
+            },
+            {
+                id: DOT_LOCATION,
+                fun: { Fungible: totalFeeInDot },
+            },
+        ],
+    }
+    const destination = { v4: parahchainLocation(assetHubParaId) }
+
+    const feeAsset = {
+        v4: DOT_LOCATION,
+    }
+    const customXcm = buildAssetHubPNATransferFromParachain(
+        parachain.registry,
+        ethChainId,
+        sourceAccount,
+        beneficiaryAccount,
+        asset.locationOnAH,
+        asset.locationOnEthereum,
+        messageId
+    )
+
+    return parachain.tx.polkadotXcm.transferAssetsUsingTypeAndThen(
+        destination,
+        assets,
+        "Teleport",
+        feeAsset,
+        "DestinationReserve",
+        customXcm,
+        "Unlimited"
+    )
+}
+
+function createAssetHubTxForPNAFromForeignConsensus(
+    parachain: ApiPromise,
+    ethChainId: number,
+    beneficiaryAccount: string,
+    amount: bigint,
+    asset: Asset
+): SubmittableExtrinsic<"promise", ISubmittableResult> {
+    const assets = {
+        v4: [
+            {
+                id: asset.location,
+                fun: { Fungible: amount },
+            },
+        ],
+    }
+    const feeAsset = {
+        v4: asset.location,
```

Comment:
Here we implicitly swap "KSM" (or any other bridged asset) for DOT during execution. The UI doesn't really inform the user of the swap and the cost implications associated with it. So let's pay with DOT for this release. It is ok to hardcode DOT as the fee here.

Reply:
There is no swap in this case. It's for transfer KSM from AH to Ethereum. The fee asset is same as the PNA in V1.

Reply:
Ah ok my mistake, the issue is with my understanding is that when asset hub is the source, the delivery fee is taken directly from the users DOT balance and not from inside the XCM holding/fee register.